runtime/lua/vim/lsp/util.lua

```diff
@@ -211,7 +211,7 @@ function M.apply_text_edits(text_edits, bufnr)
   local lines = api.nvim_buf_get_lines(bufnr, start_line, finish_line + 1, false)
   local fix_eol = api.nvim_buf_get_option(bufnr, 'fixeol')
   local set_eol = fix_eol and api.nvim_buf_line_count(bufnr) <= finish_line + 1
-  if set_eol and #lines[#lines] ~= 0 then
+  if set_eol and (#lines == 0 or #lines[#lines] ~= 0) then
```

Comment:
can we just put this inside of the `if` block with a comment about what's happening maybe?

```lua
if set_eol then
  if ...
  end
end
```

Reply:
You could also maybe do `#(lines[#lines] or "")` maybe as well to make it more obvious? I'm not super sure what way is more readable.

Reply:
You mean like this?

```lua
  if set_eol then
    if #lines == 0 then
      table.insert(lines, '')
    elseif #lines[#lines] ~= 0 then
      table.insert(lines, '')
    end
  end
```



`#(lines[#lines] or "")` would behave differently than `(#lines == 0 or #lines[#lines] ~= 0)`

Reply:
I thought the problem was basically `lines = {}`, so `#lines = 0`, and `#lines[0] => #(nil) => error`.

So my suggestion above was to make sure we never try and get the length of a nil item.

Sorry, just trying to make sure I understand what went wrong. Thanks :)

Reply:
> I thought the problem was basically lines = {}, so #lines = 0, and #lines[0] => #(nil) => error.

Yes, that is correct - in that it avoids the error, but it also doesn't add the newline.

`#(lines[#lines] or "") ~= 0` → `#("") ~= 0` → `0 ~= 0` → `false`.

If the newline isn't added, there would be another error further below.